### PR TITLE
feat: refine group (fix#287)

### PR DESCRIPTION
**Pull request by zyy7259**
**Summary**

- isolate a group's open status with `isolated: true`
- config a isolated group's initial open status with `initialIsolatedOpen: true`
- config a group's sidebarDepth with `sidebarDepth: 2`
- nested sidebar groups

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [x] Feature
- [x] Code style update
- [x] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot: 

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

os: Mac High Sierra 10.13.6

- [x] Chrome Version 67.0.3396.99 (Official Build) (64-bit)
- [X] Firefox 60.0.2 (64-bit)
- [x] Safari Version 11.1.2 (13605.3.8)
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Comment by zyy7259** [comment]
Test demo urls:

- https://zyy7259.github.io/test-vuepress-nested-group/nested/
- https://zyy7259.github.io/test-vuepress-nested-group/zh/nested/

If you want to build the demo from scratch:

- `git clone `git@github.com:zyy7259/vuepress.git`
  - `cd vuepress`
  - `yarn`
- `cd ..`
- `git clone git@github.com:zyy7259/test-vuepress-nested-group.git`
  - `cd test-vuepress-nested-group`
  - `yarn`
  - `git checkout feature-refine-group`
  - `npm run dev`

**Comment by zyy7259** [comment]
>What does this mean / do? ...

@davestewart Right now, when you open a group, other groups will be closed automatically. With `isolated: true` a group won’t be closed when you open another group at the same level.

**Comment by marceloavf** [comment]
Looking forward for this, need any help?

**Comment by zyy7259** [comment]
Rebased from master

**Comment by jackpope** [comment]
@zyy7259 Looks like #815 was merged. Do you need anything on this?

**Comment by zyy7259** [comment]
Rebased on top of master.

File: packages/docs/docs/theme/default-theme-config.md [hunk 1]
```diff
@@ -329,7 +335,7 @@ module.exports = {
 ```
 
 ::: warning Note
-Unlike the [built-in search](#built-in-search) engine which works out of the box, [Algolia DocSearch](https://community.algolia.com/docsearch/) requires you to submit your site to them for indexing before it starts working. 
+Unlike the [built-in search](#built-in-search) engine which works out of the box, [Algolia DocSearch](https://community.algolia.com/docsearch/) requires you to submit your site to them for indexing before it starts working.
```
[the remark this answers is not in the document]
Comment (zyy7259) [reply packages/docs/docs/theme/default-theme-config.md]: Oh, I have config my IDE to trim trailing whitespace. Should I add the whitespace back?

**Comment by ulivz** [comment]
Thanks for the great work, but it's not recommended to bundle two many features in a PR.

I just create a PR containing following `necessary` features:

1. Configurable sidebarDepth for sidebar group;
2. Nested sidebar groups;
3. Clickable heading for sidebar groups.

For `isolated` and `initialIsolatedOpen`, they are not so necessary, but will bring much extra learning costs, therefore we are not going to introduce these features.

Let's move #1257 to continue.

**Pull request closed without merging**